apps/wallet/src/utils/auth/signIn.ts

```diff
@@ -81,6 +82,25 @@ const signInWithPasscode = async (
 		}
 	}
 
+	if (status === ThresholdResult.Ready) {
+		let { userAccount } = await qlClient.request<{
+			userAccount: Account | undefined;
+		}>(queries.userAccount);
+
+		if (!userAccount) {
+			const { registerAccountWithoutKey: account } = await qlClient.request<{
+				registerAccountWithoutKey: Account;
+			}>(mutations.registerAccountWithoutKey);
+
+			userAccount = account;
+		}
+
+		if (!userAccount.identifier) {
+			handleInitFail?.();
+			return;
+		}
```

Comment:
1. prefer putting this block code into a new function `registerOldAccount`
2. if in this context, sign in flow, you can get the recovery key and register to api after they input the recovery key

Reply:
You're right, as we have asked for the recovery seed & reconstruct the `tkey` so we can call `initAndRegisterWallet` directly, we don't need to use the `registerWithoutKey` anymore. Do you want to revert some commits related to the `registerWithoutKey` in `gotenks` also (I'll remove it from `walless` also)?

Reply:
let's do it :v